cmd/client/proxy/publish.go

```diff
@@ -0,0 +1,127 @@
+package proxy
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/remotedialer"
+	"github.com/rancher/wins/cmd/cmds/flags"
+	"github.com/rancher/wins/pkg/defaults"
+	"github.com/rancher/wins/pkg/npipes"
+	"github.com/rancher/wins/pkg/panics"
+	"github.com/rancher/wins/pkg/proxy"
+	"github.com/urfave/cli"
+)
+
+var _proxyFlags = []cli.Flag{
+	cli.GenericFlag{
+		Name:  "publish",
+		Usage: "[required] [list-argument] Publish a port or a range of ports, e.g.: TCP:443 TCP:80-81 (note: only TCP is supported)",
+		Value: flags.NewListValue(),
+	},
+	cli.StringFlag{
+		Name:  "proxy",
+		Usage: "[optional] Specifies the name of the proxy listening named pipe",
+		Value: defaults.ProxyPipeName,
+	},
+}
+
+var _proxyPorts []int
+
+func _proxyRequestParser(cliCtx *cli.Context) (err error) {
+	// Check if ports are provided
+	publishList := flags.GetListValue(cliCtx, "publish")
+	if publishList.IsEmpty() {
+		return fmt.Errorf("No ports to publish")
+	}
+
+	// Parse ports
+	publishPorts, err := publishList.Get()
+	if err != nil {
+		return fmt.Errorf("failed to parse --publish: %v", err)
+	}
+	ports, err := parsePublishes(publishPorts)
+	if err != nil {
+		return fmt.Errorf("failed to parse --publish %s: %v", publishPorts, err)
+	}
+	_proxyPorts = ports
+
+	return nil
+}
+
+func parsePublishes(publishPorts []string) (ports []int, err error) {
+	for _, pub := range publishPorts {
```

Comment:
A bit confused about this for loop: why do we return an error when one `pub` does not satisfy what we are looking for? Should we continue the loop.

Reply:
Because then the input for wins cli proxy is malformed, which means we should fail out or have unintended consequences.

e.g. imagine if I did
`wins cli proxy TCP:9090TCP:8080 UDP:9090`

You wouldn't want to continue with just `UDP:9090`

Reply:
Entirely agree. Thanks for explaning